.github/workflows/test-js-packages.yml

```diff
@@ -47,26 +47,37 @@ jobs:
           retention-days: 3
 
   build-noir-wasm:
+    needs: [build-noirc-abi]
     runs-on: ubuntu-latest
     timeout-minutes: 30
 
     steps:
       - name: Checkout sources
         uses: actions/checkout@v4
 
-      - name: Setup Nix
-        uses: ./.github/actions/nix
+      - name: Setup toolchain
+        uses: dtolnay/rust-toolchain@1.71.1
+
+      - uses: Swatinem/rust-cache@v2
         with:
-          github-token: ${{ secrets.GITHUB_TOKEN }}
-          nix-cache-name: "noir"
-          cachix-auth-token: ${{ secrets.CACHIXAUTHTOKEN }}
+          key: x86_64-unknown-linux-gnu
+          cache-on-failure: true
+          save-if: ${{ github.event_name != 'merge_group' }}
```

Comment:
Pretty sure this is going to cause cache issues btw as there will be a race condition on whether the nargo or wasm compiler jobs populate this cache first.

Reply:
We could use noir-wasm as key

Reply:
Changed cache key!